### PR TITLE
vtctl ApplySchema -uuid_list

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

`vtctlient ApplySchema` now support a new optional `-uuid_list` flag. It is possible for the user to explicitly specify the UUID(s) for given migration(s). UUIDs must be in a specific format. If given, number of UUIDs must match the number of DDL statements. Example:

```shell
vtctlclient OnlineDDL ApplySchema -sql "drop table t1, drop table t2" -uuid_list "d08f0000_51c9_11ec_9cf2_0a43f95f28a3,d08f0001_51c9_11ec_9cf2_0a43f95f28a3" commerce
```

Format is comma delimited, zero or more UUIDs.

Vitess will assign each migration with given UUID in order of appearance.
It is the user's responsibility to ensure given UUIDs are globally unique. If the user submits a migration with an already existing UUID, that migration never gets scheduled nor executed.

The purpose of this change is to give users better control over submitted migrations. Moreover, it is a means for users to achiever idempotency: the user can _choose_ to resubmit a migration with same UUID, intentionally ensuring it does not execute again.

`-uuid_list` is optional. If omitted or if empty then Vitess auto-generates UUIDs for the migrations.

## Related Issue(s)

- Tracking: #6926 

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by mavenraven** [comment]
@shlomi-noach looks great to me from a consumer perspective!

**Review by mavenraven — APPROVED** [review]

File: doc/releasenotes/13_0_0_summary.md [hunk 1]
```diff
@@ -60,6 +60,17 @@ For example:
 vtctlclient OnlineDDL commerce complete d08ffe6b_51c9_11ec_9cf2_0a43f95f28a3
 ```
 
+### vtctl/vtctlclient ApplySchema -uuid
```
Comment (deepthi) [inline doc/releasenotes/13_0_0_summary.md:63]: Intuitively `uuid` seems to indicate that it is a single uuid versus a list or a set. `uuids` feels awkward, but maybe `uuid_list`??

File: go/vt/vtgate/evalengine/expressions.go [hunk 2]
```diff
@@ -251,21 +251,21 @@ func (t TupleExpr) Evaluate(env *ExpressionEnv) (EvalResult, error) {
 }
 
 // Evaluate implements the Expr interface
-func (b *BindVariable) Evaluate(env *ExpressionEnv) (EvalResult, error) {
-	val, ok := env.BindVars[b.Key]
+func (bv *BindVariable) Evaluate(env *ExpressionEnv) (EvalResult, error) {
```
Comment (deepthi) [inline go/vt/vtgate/evalengine/expressions.go:254]: Changes to this file seem unrelated. Are they really required?

Comment (shlomi-noach) [reply go/vt/vtgate/evalengine/expressions.go:254]: Oh this was due to the new linter throwing errors in that file. I'll see what I can do here.

Comment (shlomi-noach) [reply go/vt/vtgate/evalengine/expressions.go:254]: ok this at least was already merged by another PR

**Comment by shlomi-noach** [comment]
flag `-uuid` renamed to `-uuid_list`

`explicitUUID` renamed to `proidedUUID` throughout the code.

**Review by deepthi — APPROVED** [review]
